include/exadg/time_integration/extrapolation_scheme.cpp

```diff
@@ -165,35 +165,20 @@ ExtrapolationConstants::set_adaptive_time_step(unsigned int const          curre
   }
 }
 
-void
-ExtrapolationConstants::update(unsigned int const current_order)
-{
-  // when starting the time integrator with a low order method, ensure that
-  // the time integrator constants are set properly
-  if(current_order <= order && start_with_low_order == true)
-  {
-    set_constant_time_step(current_order);
-  }
-  else
-  {
-    set_constant_time_step(order);
-  }
-}
-
 void
 ExtrapolationConstants::update(unsigned int const          current_order,
-                               std::vector<double> const & time_steps)
+                               std::vector<double> const & time_steps,
+                               bool const                  adaptive)
 {
   // when starting the time integrator with a low order method, ensure that
   // the time integrator constants are set properly
-  if(current_order <= order && start_with_low_order == true)
-  {
-    set_adaptive_time_step(current_order, time_steps);
-  }
-  else // adjust time integrator constants since this is adaptive time stepping
-  {
-    set_adaptive_time_step(order, time_steps);
-  }
+  unsigned int update_order =
+    (current_order <= order && start_with_low_order == true) ? current_order : order;
+
+  if(adaptive)
+    set_adaptive_time_step(update_order, time_steps);
+  else
+    set_constant_time_step(update_order);
```

Comment:
I am against such refactoring changes in a PR such as the present one. You could argue that you are a 100% sure that the code does the same as before, and I might even agree on this. However, we had cases in the past where the developers and code reviewers were proven wrong and where the code was broken afterwards for some subtle reasons. I do not want to block the present PR for this particular reason. However, (and also in light of e.g. other PRs such as PR #381) I think we need to agree for the future that refactoring of the code should be subject of separate PRs and not appear within a big PR with 1,600 new lines of code. I have seen these 7 lines of code at least 4 times in your PR. Doesn't this call for a base class that implements exactly this logic? Ideally, and as written elsewhere, we would introduce this base class in a PR prior to the present PR and adapt BDF and EXT coefficients in a first step. Afterwards, the present PR would build upon this new design.

Reply:
You asked me to refactor the code in several comments above.